src/fwd.cpp

```diff
@@ -182,7 +182,7 @@ PJ_XY pj_fwd(PJ_LP lp, PJ *P) {
 
     last_errno = proj_errno_reset(P);
 
-    if (!P->skip_fwd_prepare)
+    if (!P->skip_fwd_prepare && 0 != strcmp(P->short_name, "s2"))
```

Comment:
Why is this hack needed ? Ideally, we shouldn't need that.

Reply:
I don't understand what the purpose of `fwd/inv` `prepare/finalize` is, so I was unsure how to handle it. The S2 projection is intended to return points expressed in s,t-coordinates, which are in [0,1] x [0,1].  I observed that fwd_finalize was making the output several orders of magnitude bigger, so I wanted to disable it.   Is there a different way I can set the `skip` parameters specifically for S2?

Reply:
Ah, for a classic projection, PROJ multiplies/demultiplies the result by P->a. You could perhaps try setting P->left = PJ_IO_UNITS_PROJECTED; and P->right = PJ_IO_UNITS_PROJECTED; in the setup function in s2.cpp

Reply:
This has introduced an error of `can't initialize operations that take non-angular input coordinates`. Is there a different solution I could try?

Reply:
oops sorry, it should have been:
```
P->left = PJ_IO_UNITS_RADIANS; 
P->right = PJ_IO_UNITS_PROJECTED;
```